docker/mnt/run.sh

```diff
@@ -0,0 +1,121 @@
+#! /bin/bash
+set -e
+
+if [ -z "$1" ]; then
+    # the main container need not stay running for services
+    exit 0
+fi
+
+function setup() {
+    [ -n "$1" ] && TEST="$1"
+
+    if [[ "$TEST" =~ ^python ]]; then
+
+        scripts/uninstall-requirements.sh
+        pip install \
+            -r requirements/requirements.txt \
+            -r requirements/dev-requirements.txt \
+            coveralls
+
+        # some kind of optimization?
+        # skip Running setup.py bdist_wheel for ... ?
+        rm -rf /root/.cache/pip
+
+        /usr/lib/jvm/jdk1.7.0/bin/keytool -genkey \
+            -keyalg RSA \
+            -keysize 2048 \
+            -validity 10000 \
+            -alias javarosakey \
+            -keypass onetwothreefourfive \
+            -keystore InsecureTestingKeyStore \
+            -storepass onetwothreefourfive \
+            -dname 'CN=Foo, OU=Bar, O=Bizzle, L=Bazzle, ST=Bingle, C=US'
+    fi
+
+    if [ "$TEST" = "javascript" -o "$JS_SETUP" = "yes" ]; then
+        npm install
+        bower install --config.interactive=false
+    fi
+
+    /mnt/wait.sh
+
+    if [[ "$TEST" =~ ^python ]]; then
+        su cchq -c "./manage.py create_kafka_topics"
+    fi
+}
+
+function run_tests() {
+    TEST="$1"
+    if [ "$TEST" != "javascript" -a "$TEST" != "python" -a "$TEST" != "python-sharded" ]; then
+        echo "Unknown test suite: $TEST"
+        exit 1
+    fi
+    shift
+    setup $TEST
+    su cchq -c "../run_tests $TEST $(printf " %q" "$@")"
```

Comment:
A but confused by the paths here. From what I can make out:
* This file is `/mnt/run.sh` 
* HQ source is in `/mnt/commcare-hq-ro`
* The `_run_tests` function get's written to `/mnt/run_tests` below
* `/mnt/lib` - not too sure what's in there or how HQ knows about it

Still trying to get head around the setup. Perhaps a quick overview would be helpful?